### PR TITLE
Support Swift 2.0

### PR DESCRIPTION
**Pull request by modocache**
1. Run Edit > Convert > To Latest Swift Syntax.
2. beEmpty() matcher requires an additional definition, not just for
   NSString but for String as well. beEmpty() callsites required casting
   in order to resolve ambiguities around which function was being
   called.

### CODE REVIEW

**Comment by modocache** [comment]
@Quick/contributors This obviously won't pass CI, and probably won't for a while, until CIaaS companies like Travis CI and Circle CI support Xcode 7. 

Just as when we supported Swift 1.2, I suggest having `master` compatible with the latest toolchain. We can create a `swift-1.2` branch for backwards compatibility. I'll add a commit to this branch that disables CI and points the README CI badge at the state of the `swift-1.2` branch.

Thoughts? :raised_hands:

**Comment by paulyoung** [comment]
Looks pretty good to me :star2:

**Comment by jeffh** [comment]
Sounds good to me!

File: Nimble/Matchers/RaisesException.swift [hunk 1]
```diff
@@ -10,14 +10,14 @@ import Foundation
 /// nil arguments indicates that the matcher should not attempt to match against
 /// that parameter.
 public func raiseException(
-    named: String? = nil,
+    named named: String? = nil,
```
Comment (ashfurrow) [inline Nimble/Matchers/RaisesException.swift]: Shouldn't the parameter name internal to the function be called `name`?

**Comment by fabiomassimo** [comment]
My two cents: the adoption of Swift 2 sounds like a major migration and it's available through Xcode 7 (beta). Wouldn't it be better to work on a separate branch `swift-2.0` and keep `master` up to date with current Xcode stable version?
It wouldn't be weird if someone is expecting that `master` branch works with current stable version of Xcode rather than a pre-release one.

**Comment by ashfurrow** [comment]
Looks good Brian!

@fabiomassimo I think keeping this PR open until Swift 2 is out would be a good idea. For further updates (like Swift 2 beta 2), we can submit PRs _into_ this branch. 

That's just one idea – lots of different approaches.

**Comment by modocache** [comment]
@fabiomassimo This confusion definitely occurred during the period of time we supported Swift 1.1 and Swift 1.2. At that time, we used a `swift-1.1` branch. Tons of issues were opened by people who were confused by the fact that Quick "didn't compile".

I'm open to new ideas this time. For context, here's our previous discussion on the subject: https://github.com/Quick/Quick/issues/243

File: Nimble/Matchers/RaisesException.swift (hunk 1 above)
Comment (modocache) [inline Nimble/Matchers/RaisesException.swift]: I believe this is meant to read "*expect* this closure *to* *raise* an *exception* *named*...", and thus the name of this parameter.

In any case, that's the current API. Changing it should be part of a separate pull request.

Comment (ashfurrow) [inline Nimble/Matchers/RaisesException.swift]: Ah, you're right. I was suggesting keeping the interface the same but renaming the local parameter to "name" instead of "named" – does that make sense? Totally up for a separate PR.

**Comment by neilpa** [comment]
Note that calling the branch `swift-1.2` won't be [compatible with carthage](https://github.com/Carthage/Carthage/issues/469#issuecomment-107951691)

**Comment by modocache** [comment]
@neilpa Looks like the issue is with *tags* containing dashes and hyphens, not *branches*. We'll be tagging versions using [semantic versioning](http://semver.org/), which Carthage will certainly support, so no worries there! :+1: 

@jeffh We're probably going to want to tag a Swift 2.0-compatible release once we cut a `swift-1.2` branch. The latest version of Nimble is `v1.0.0-rc.1`, which supports Swift 1.2. What should the Swift 2.0-compatible version be? There are no API changes, but the new code won't run on Xcode 6.*. Personally, I think it should be `v2.0.0-rc.1` (although to be honest, I'm not a fan of `rc` releases). Thoughts? :8ball:

File: Nimble/Matchers/RaisesException.swift (hunk 1 above)
Comment (modocache) [inline Nimble/Matchers/RaisesException.swift]: Ah, gotcha. Yeah, open an issue if you feel strongly about it and we'll deal with it separately. :snail:

**Comment by modocache** [comment]
Closing in favor of #132. Let's continue this discussion there.

(Boy, it sure would be nice if GitHub allowed you to edit the target of a pull request.)
